visitor.php

```diff
@@ -305,8 +306,8 @@ public function enterNode(Node $node)
 
     private static function getNodeName(Node $node): string
     {
-        if (($node instanceof Function_) || ($node instanceof ClassMethod)) {
-            return $node->name->name;
+        if (($node instanceof Function_) || ($node instanceof ClassMethod) || ($node instanceof Class_)) {
+            return $node->name->name ?? '';
```

Comment:
Here we go 🐎 
```suggestion
        if (($node instanceof Function_) || ($node instanceof ClassMethod) || ($node instanceof Class_ && isset($node->name))) {
            return $node->name->name;
```

Reply:
`Cannot access property $name on PhpParser\Node\Identifier|null. `

Reply:
I need to understand how it works.
A PHPStan "OK" may be a bug!

Reply:
Trying to find the root cause with `\PHPStan\dumpType();`

Reply:
`&& !$node->isAnonymous()` would go for brave&potent boys but I'm looking for a solution to satisfy static analysis too.

Reply:
`isAnonymous` is inside phpstan.phar so cannot be analyzed. Whyyy???

Reply:
congratulations, you might have found a PHPStan bug 😅
https://phpstan.org/r/e29b76ac-5398-4505-b64a-b568c9c37c53

Reply:
Reported in https://github.com/phpstan/phpstan/issues/8968

Reply:
Changing to

```php
        if ((($node instanceof Function_) || ($node instanceof ClassMethod) || ($node instanceof Class_)) && $node->name !== null) {
            return $node->name->name;
        }
```